### ISSUE TITLE
Add HTTP Headers for fetching an image via URL

### ISSUE THREAD

**Issue by ihr** [h2non/imaginary#85]
Hi,

I could not find a way to tell imaginary to fetch an image from URL while using Authorization header. It would be useful to be able to add any type of header

**Comment by h2non** [comment]
This is simply not supported and expected. 

What are you looking here? Having a pre-defined constant authorization token defined at server boot time or dynamically define an authorization header per request that should be forwarded to the image origin server?

**Comment by ihr** [comment]
Hi there,

Thanks for the answer. Both, static & dynamic setting of a Authorization header makes sense. However, dynamic would make imaginary much more interoperable.

Cheers!

**Comment by h2non** [comment]
Can you make a concrete proposal? I would not mind supporting it.

**Comment by ihr** [comment]
For a static implementation, a _- H_ flag (similar to curl) which can be repeated more than once would be enough. For a dynamic approach, it would be better to use a _X-Forward-<Header_Name>_. As an example:
```bash
curl -O "http://localhost:8088/crop?width=500&height=400&url=https://raw.githubusercontent.com/h2non/imaginary/master/fixtures/large.jpg" -H "X-Forward-Authorization: Basic Ym9zY236Ym9zY28=" 

Cheers!

**Comment by h2non** [comment]
X-Fordward-Authorization looks good to me.

**Comment by h2non** [comment]
@ihr I finally had time to work on this. You can check implementation details here in this PR:
https://github.com/h2non/imaginary/pull/89

and more specifically this commit:
https://github.com/h2non/imaginary/pull/89/commits/b7041a1ee9fb9083081488e94131c7a5b3baedc5

**Comment by h2non** [comment]
Feature shipped in `v0.1.26`, also available as Docker release. Closing.
